### PR TITLE
Cross platform external command capability

### PR DESCRIPTION
**Pull request by wpferguson**
Added system library to handle running external commands on different operating systems in a transparent manner.  Added file and path quoting that is cross platform compatible.  Changed gimp.lua to use the new utilities.

### CODE REVIEW

File: contrib/gimp.lua [hunk 1]
```diff
@@ -104,7 +105,7 @@ local function sanitize_filename(filepath)
 
   return path .. sanitized .. "." .. filetype
 end
-
+]]
```
Comment (supertobi) [inline contrib/gimp.lua]: Please delete the unused code.

**Comment by supertobi** [comment]
I don't have MacOs or Windows for testing this. I put my trust in you that it works.
Perhaps you can add an example script to show how to write a multi OS script.

File: contrib/gimp.lua (hunk 1 above)
Comment (wpferguson) [reply contrib/gimp.lua]: Thanks

**Comment by supertobi** [comment]
> Removed leftover unused code 

Wow, that was fast. 👍

**Comment by wpferguson** [comment]
I have an idea for an example script.  Lots of comments with a little bit of code :-) and it will actually do something useful.

**Comment by wpferguson** [comment]
@supertobi , just tested a new script, written with these changes, on linux, windows and macos without any problems.  I think it's ready to go.

**Comment by supertobi** [comment]
Then push it.

**Pull request merged**